test/e2e/conftest.py

```diff
@@ -30,24 +33,202 @@ def pytest_sessionstart(session):
+    """
+    if xdist.is_xdist_worker(session):
+        return
+
+    with open(_test_session_config_file_path(), "a") as f:
+        f.write(
+            """[STATE]
+is_finished=True
+"""
+        )
+
+
+def _cleanup_test_session_config() -> None:
+    config_path = _test_session_config_file_path()
+    if os.path.exists(config_path):
+        os.remove(config_path)
+
+
+def _is_test_session_finished() -> bool:
+    """
+    Since we parallelize tests via xdist and fixture tear downs run after
+    pytest_sessionfinish hook, we need this inter process communication via
+    file
+    """
+    config = configparser.ConfigParser()
+    if not config.read(_test_session_config_file_path()):
+        return False
+
+    return config.get("STATE", "is_finished", fallback="False") == "True"
+
+
+def _write_organization_account_to_test_session_config(org_account):
+    with open(_test_session_config_file_path(), "w") as f:
+        f.write(
+            f"""
+[ORGANIZATION_ACCOUNT]
+id={str(org_account.id)}
+name={org_account.name}
+"""
+        )
+
+    account_to_verify = _read_organization_account_from_test_session_config()
+    assert account_to_verify
+    assert account_to_verify.id == org_account.id
+    assert account_to_verify.name == org_account.name
+
+
+def _read_organization_account_from_test_session_config() -> Optional[Account]:
+    """
+    Returns None if there is no organization account for this session.
+    One reason being that it got deleted already by a fixture cleanup.
+    """
+    config = configparser.ConfigParser()
+    if not config.read(_test_session_config_file_path()):
+        return None
+
+    return Account(
+        id=uuid.UUID(config.get("ORGANIZATION_ACCOUNT", "id")),
+        name=config.get("ORGANIZATION_ACCOUNT", "name"),
+    )
+
+
+def _test_session_config_file_path() -> str:
+    return DEFAULT_LAYER_PATH / "e2e_test_session.ini"
+
+
+async def create_organization_account() -> Account:
+    config = await ConfigManager().refresh()
+    client = LayerClient(config.client, logger)
+    org_account_name, display_name = pseudo_random_account_name()
+    account = client.account.create_organization_account(org_account_name, display_name)
+
+    # We need a new token with permissions for the new org account
+    # TODO LAY-3583 LAY-3652
+    await ConfigManager().refresh(force=True)
+
+    return account
+
+
+@pytest.fixture(autouse=True)
+def _test_session_teardown(
+    initialized_organization_account: Account, client: LayerClient
+) -> Iterator:
+    """
+    Invoked on every test teardown.
+
+    Since we parallelize tests, and fixture teardowns happen after pytest_sessionfinish hook, we need
+    this autoUse fixture to clean up the global test session.
+    """
+    yield
+    session_finished = _is_test_session_finished()
+    if not session_finished:
+        return
+    _cleanup_organization_account(client)
+    _cleanup_test_session_config()
```

Comment:
Are we sure this will get executed at least once?

Reply:
it should yes since the flag is set by `pytest_sessionfinish` with '`a`' append mode

what could happen though is that `pytest_sessionfinish` suddenly starts being executed after fixture teardowns, so i can add an attempt to delete the org account also inside the hook

Reply:
Are fixtures cleaned up after every test or at the end of the run?

What will happen if the tests get split like this:

w0 - orgTest, regularTest
w1 - orgTest, regularTest

?

If the tear down happens after `orgTest`, that will be before `pytest_sessionfinish` so we will never clean up.

Reply:
i dont understand the question sorry. currently there is an autoUse fixture that will be run after every test, so one single test teardown will be the one to execute the session cleanup

Reply:
oh, I see, so that cleanup is attempted after every test regardless of whether it actually uses the org or not.

Reply:
yes indeed, this is why I call it `_test_session_teardown`, kinda like our own hook for our distributed use case.

i thought `pytest_sessionfinish` was that but it isnt behaving as i expected

Reply:
using this https://github.com/pytest-dev/pytest/issues/7484 now